projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -444,6 +445,10 @@ public function render_block( $attributes ) {
 			'size'            => ! empty( $attributes['postsToShow'] ) ? absint( $attributes['postsToShow'] ) : 3,
 		);
 
+		if ( isset( $attributes['displayHeadline'] ) && true === $attributes['displayHeadline'] ) {
+			$block_attributes['headline'] = $content;
```

Comment:
To be honest, overwriting this feels weird since the `content` could change in the future and no longer represent only the `headline`. However, based on what this block does, and what it has done, I don't know that it's a big risk.

Reply:
It does indeed feel a bit fragile, and maybe a bit confusing in the future. We already have a complex setup with  `headline` and `showHeadline` for the PHP rendering of related posts, and I'm afraid this will only complicate things further. I cannot really think of a better solution though.

Reply:
Yeah, agreed. Open to suggestions.